### DISCUSSION TITLE
iPhone app: very slow loading of 4k video

### DISCUSSION THREAD

**Discussion by siagimir** [immich-app/immich#848] [Q&A] [answered]
Hello everyone,
please, does anyone have a problem with a very slow loading of 4k (60fps) HDR videos in the mobile app?
4k - 30 fps video is buffering at an "okay-ish" level (but it's reached its limit I'd say). 

On the web (Chrome), there is no problem with viewing all types of videos.

Videos have been recorded on iPhone 12 Pro.

Thanks!

**Comment by alextran1502** [comment]
Are you accesssing from the local network or through dns? What is your server spec? What is your network spec?

**Reply by siagimir** [reply]
I am accessing from my local network. 5GHz ac WiFi, both iPhone and PC/Mac (where I use Chrome).

Regarding DNS, I use Pi-hole. The server's specs are: Intel® Core™ i3-10110U Processor, 8 GB RAM with the main 128 SSD, photos are stored in 2 TB HDD via M2 SATA. 

When loading a video, there is no spike or significant usage of the server's CPU (a few percent) - checked via docker stats or htop.

When I load a video in Chrome, there's the spike in the network (first green "circle"). When I load the same video in the mobile app, there's almost no spike - just a little one (the second green "circle").

<img src="https://user-images.githubusercontent.com/7303508/197363584-34825466-6caa-42ff-aca1-e336ca1f1b36.jpg" width=40% height=40%>

I also checked my iPhone's capability regarding the download speed from the web - it's okay. But I'll somehow try to check the speed between the phone and the server.

**Reply by siagimir** [reply]
I have a few hypotheses so far. It looks like the mobile has trouble reaching a higher download speed. 
So far thank you for the reply.

**Reply by alextran1502** [reply]
Hello, my assumption is that is might be disk issue. I tried recording a 4K 60FPS and access the server using my local network, the video loaded and buffered normally on my end. I think the only different from my system to yours is the disk setup, so it might be it.

**Reply by siagimir** [reply]
[moved this post to the new thread and marked as the answer]

**Comment by siagimir** [comment, marked as answer]
In the end, it looked like the mobile's WiFi signal strength issue. Loading and buffering were good when I came much closer (1 m) to my router (lol).
But such a thing has never happened to me when using Google Photos with the same video and conditions. I understand I cannot compare this to such a huge company's product, but maybe I'll try to look at the encoding process which could be maybe optimised to a lower bitrate in the future.
Anyway, thank you very much for responding.

**Reply by alextran1502** [reply]
That is kinda strange though. I've tested the streaming performance when I was close to the router vs when I was at the opposite end and I didn't notice a lot of buffering. 

I know there can be many different factors but the issue you are experiencing will be a good issue to open for optimize streaming video for the app.

**Reply by siagimir** [reply]
It is strange, indeed. But as far as I can remember, I've always had small problems with the strength of WiFi at home.

The optimisation would be great, though. Would you like me to create an issue for it?
